app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
+function findMatchingClosingBracketIndex(source, fromIndex) {
+  let result = -1
+  let currentDepth = 1
+  for (let i = fromIndex; i < source.length; i++) {
+    const char = source[i]
+    if (char === "{") {
+      currentDepth++
+    } else if (char === "}") {
+      currentDepth--
+      if (currentDepth <= 0) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
```

Comment:
This is a bit nitpicky, but instead of calling it `source` can we call it `joinedItems` to stay consistent with other functions?

Reply:
_Or_ we could use `source` on the other functions too 😁.
Honestly whichever way is fine for me 👍